web_src/js/features/common-global.js

```diff
@@ -443,3 +444,24 @@ export function checkAppUrl() {
   showGlobalErrorMessage(`Your ROOT_URL in app.ini is "${appUrl}", it's unlikely matching the site you are visiting.
 Mismatched ROOT_URL config causes wrong URL links for web UI/mail content/webhook notification/OAuth2 sign-in.`);
 }
+
+/**
+ * Allows to fold anywhere with the correct classes
+ *
+ * @example
+ * ```html
+ * <div class="fold-container">
+ *    <div class="header">
+ *      <button class="fold-button">Fold</button>
+ *    </div>
+ *    <div class="fold-body">
+ *      Show foldable content
+ *    </div>
+ * </div>
+ * ```
+ */
+export function initGlobalFolding() {
+  $(document).on('click', '.fold-container .fold-button', ({currentTarget}) => {
+    invertFileFolding(currentTarget.closest('.fold-container'), currentTarget);
+  });
+}
```

Comment:
Can we use a [details](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) for the toggle mechanism? That works natively without any JS and I know it can be (ab)used for a lot of use cases.